src/dos/drive_local.cpp

```diff
@@ -87,21 +117,11 @@ bool localDrive::FileOpen(DOS_File** file, char * name, Bit32u flags) {
 	CROSS_FILENAME(newname);
 	dirCache.ExpandName(newname);
 
-	//Flush the buffer of handles for the same file. (Betrayal in Antara)
-	Bit8u i,drive=DOS_DRIVES;
-	localFile *lfp = nullptr;
-	for (i = 0; i < DOS_DRIVES; i++) {
-		if (Drives[i] == this) {
-			drive = i;
-			break;
-		}
-	}
-	for (i = 0; i < DOS_FILES; i++) {
-		if (Files[i] && Files[i]->IsOpen() && Files[i]->GetDrive()==drive && Files[i]->IsName(name)) {
-			lfp=dynamic_cast<localFile*>(Files[i]);
-			if (lfp) lfp->Flush();
-		}
-	}
+	// If the file's already open then flush it before continuing
+	// (Betrayal in Antara)
+	auto open_file = dynamic_cast<localFile *>(FindOpenFile(this, name));
```

Comment:
Ugh, downcasting :(

But the dynamic cast was in this code before already, so there's nothing we can do about this right now, and there's a nullcheck below anyway so it's fine.